modules/loader.js

```diff
@@ -559,6 +567,131 @@ __private.loadBlockChain = function() {
+
+	modules.blocks.utils.loadBlockByHeight(
+		blockToVerify.height - 1,
+		(lastBlockError, secondLastBlockToVerify) => {
+			if (lastBlockError) {
+				return setImmediate(cb, lastBlockError);
+			}
+
+			// Set the block temporarily for block verification
+			modules.blocks.lastBlock.set(secondLastBlockToVerify);
+			const result = modules.blocks.verify.verifyBlock(blockToVerify);
+
+			// Revert last block changes
+			modules.blocks.lastBlock.set(lastBlock);
+
+			if (result.verified) {
+				return setImmediate(cb, null);
+			}
+			return setImmediate(cb, result.errors);
+		}
+	);
+};
+
+/**
+ * Validate own block chain before startup
+ *
+ * @private
+ * @param {function} cb
+ * @returns {setImmediateCallback} cb, err
+ */
+__private.validateOwnChain = cb => {
+	const currentBlock = modules.blocks.lastBlock.get();
+	const currentHeight = currentBlock.height;
+	const currentRound = slots.calcRound(currentHeight);
+
+	// Starting height should be end height of second last round
+	let startHeight;
+	if (currentRound === 1) {
+		// Skip the genesis block validation
+		startHeight = 2;
+	} else {
+		startHeight = slots.calcRoundEndHeight(currentRound - 2);
+	}
+
+	async.series(
+		[
+			// Validate the top most block
+			seriesCb => {
+				library.logger.info(
+					`Validating current block with height ${currentHeight.height}`
+				);
+
+				__private.validateBlock(currentBlock, validateBlockErr => {
+					if (validateBlockErr) {
+						// Current block is invalid move to next step in async.series
+						return setImmediate(seriesCb, null);
+					}
+					// Current block is valid don't need to do any thing exit async.series
+					return setImmediate(seriesCb, true);
+				});
+			},
+
+			// Validate last block of second last round
+			seriesCb => {
+				library.logger.info(
+					`Validating last block of second last round with height ${startHeight}`
+				);
+
+				modules.blocks.utils.loadBlockByHeight(
+					startHeight,
+					(lastBlockError, startBlock) => {
+						__private.validateBlock(startBlock, validateBlockErr => {
+							if (validateBlockErr) {
+								library.logger.error(
+									`There are more than ${currentHeight -
+										startHeight} invalid blocks. Can't delete those to recover the chain.`
+								);
+								return process.emit(
+									'cleanup',
+									'Your block chain is invalid. Please rebuild from snapshot.'
+								);
+							}
+							// Now move to next step
+							return setImmediate(seriesCb, null);
+						});
+					}
+				);
+			},
+
+			// Delete all invalid blocks in that range
+			seriesCb => {
+				async.doDuring(
+					doWhilstCb => {
+						modules.blocks.chain.deleteLastBlock(doWhilstCb);
+					},
+					(deleteLastBlockStatus, testCallBb) => {
+						__private.validateBlock(
+							modules.blocks.lastBlock.get(),
+							validateError => setImmediate(testCallBb, null, !!validateError)
+						);
+					},
+					doDuringErr => setImmediate(seriesCb, doDuringErr)
+				);
+			},
+		],
+		() => {
```

Comment:
No error handling in case some block fails verification? Or the `true` value we are passing as error in first series function?

Reply:
We reach to this callback by explicitly on need. Only way to do is to send an error in previous steps. So we explicitly ignored the errors here.